code/Controllers/HomeController.cs

```diff
@@ -42,24 +43,41 @@ public HomeController(BagsApi bagsApi)
 			{
 				var product = await bagsApi.GetProduct(productId ?? 0);
 
-				ViewData["url"] = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+				model.url = $"https://bagcupid.com/app/{path.TrimLeadingSlash()}";
 
-				if (product.Images != null && product.Images.Count() > 0)
-					ViewData["image"] = product.Images
+				if (product?.Images != null && product.Images.Count() > 0)
+					model.image = product.Images
 						.Aggregate((selectedImage, nextImage) => (nextImage.Priority < selectedImage.Priority) ? nextImage : selectedImage)
 						.Large;
 			}
 
-			return View("index");
+			return View("index", model);
 		}
 	}
 
+	public class MetaViewModel
+	{
+		public String url { get; set; }
+		public String type { get; set; }
+		public String title { get; set; }
+		public String description { get; set; }
+		public String image { get; set; }
+	}
+
 	public static class Extensions
 	{
 		public static UInt64? TryParseUInt64(this String value)
 		{
 			UInt64 id = 0;
 			return UInt64.TryParse(value, out id) ? id : (UInt64?)null;
 		}
+
+		public static String TrimLeadingSlash(this String value)
+		{
+			if(value.StartsWith("/"))
```

Comment:
Style: This check is unnecessary, `TrimStart('/')` will trim all leading `/` off of the string, but if the string doesn't have a leading `/` character it will just return the original string.  Doing the `StartsWith` check adds complexity (along with this extension method) without providing value, so recommend just doing `path.TrimStart('/')`.